app/javascript/components/components/Header.js

```diff
@@ -18,27 +19,27 @@ class Header extends Component {
         <Nav>
           {!logged_in &&
             <NavItem>
-              <a href="/" className="nav-link">Home</a>
+              <NavLink to="/" className="nav-link">Home</NavLink>
             </NavItem>
           }
           {!logged_in &&
             <NavItem>
-              <a href="/apartmentindex" className="nav-link">Available Apartments</a>
+              <NavLink to="/apartmentindex" className="nav-link">Available Apartments</NavLink>
             </NavItem>
           }
           {logged_in &&
             <NavItem>
-              <a href={sign_out_route} className="nav-link">Sign Out</a>
+              <NavLink to={sign_out_route} className="nav-link">Sign Out</NavLink>
```

Comment:
This is an odd one because you have a mix of routing to outside pages and internal react routing. Each one is handled differently. Routes to external pages like Devise do need to be an `a href` while internal routing to other React components needs to be `NavLink to`.